centreon-certified/clickhouse/clickhouse-metrics-apiv2.lua

```diff
@@ -0,0 +1,509 @@
+-- Centreon Broker clickhouse Connector Events
+--------------------------------------------------------------------------------
+
+
+-- Libraries
+local curl = require "cURL"
+local sc_common = require("centreon-stream-connectors-lib.sc_common")
+local sc_logger = require("centreon-stream-connectors-lib.sc_logger")
+local sc_broker = require("centreon-stream-connectors-lib.sc_broker")
+local sc_event = require("centreon-stream-connectors-lib.sc_event")
+local sc_params = require("centreon-stream-connectors-lib.sc_params")
+local sc_macros = require("centreon-stream-connectors-lib.sc_macros")
+local sc_flush = require("centreon-stream-connectors-lib.sc_flush")
+local sc_metrics = require("centreon-stream-connectors-lib.sc_metrics")
+
+--------------------------------------------------------------------------------
+-- Classe event_queue
+--------------------------------------------------------------------------------
+
+--------------------------------------------------------------------------------
+-- Classe event_queue
+--------------------------------------------------------------------------------
+
+local EventQueue = {}
+EventQueue.__index = EventQueue
+
+--------------------------------------------------------------------------------
+---- Constructor
+---- @param conf The table given by the init() function and returned from the GUI
+---- @return the new EventQueue
+----------------------------------------------------------------------------------
+
+function EventQueue.new(params)
+  local self = {}
+
+  local mandatory_parameters = {
+    "user",
+    "password",
+    "http_server_url"
+  }
+
+  self.fail = false
+
+  -- set up log configuration
+  local logfile = params.logfile or "/var/log/centreon-broker/clickhouse-metrics.log"
+  local log_level = params.log_level or 1
+  
+  -- initiate mandatory objects
+  self.sc_logger = sc_logger.new(logfile, log_level)
+  self.sc_common = sc_common.new(self.sc_logger)
+  self.sc_broker = sc_broker.new(self.sc_logger)
+  self.sc_params = sc_params.new(self.sc_common, self.sc_logger)
+  
+  -- checking mandatory parameters and setting a fail flag
+  if not self.sc_params:is_mandatory_config_set(mandatory_parameters, params) then
+    self.fail = true
+  end
+  
+  --params.max_buffer_size = 1
+  
+  -- overriding default parameters for this stream connector if the default values doesn't suit the basic needs
+  self.sc_params.params.user = params.user
+  self.sc_params.params.password = params.password
+  self.sc_params.params.http_server_url = params.http_server_url
+  self.sc_params.params.clickhouse_database = params.clickhouse_database or "centreon_stream"
+  self.sc_params.params.clickhouse_table = params.clickhouse_table or "metrics"
+  self.sc_params.params.accepted_categories = params.accepted_categories or "neb"
+  self.sc_params.params.accepted_elements = params.accepted_elements or "host_status,service_status"
+  self.sc_params.params.max_buffer_size = params.max_buffer_size or 1000
```

Comment:
Is there a reason for setting to 1000 rather than 30 (found in some other SC) ?

Reply:
Sometimes we found the recommended bulk size on other tools documentations.
This was the case for clickhouse. When such situation happen, we also state in the documentation some recommended values